### PR TITLE
fix(elastic): better support for logging raised and caught exceptions and fix formatter crash

### PR DESCRIPTION
**Pull request by bvobart**
Hi again :)

When using the Elastic logger in my work project, I noticed that the logging flow for caught exceptions is not very smooth, as I first have to call `Logger.metadata(crash_reason: e)` (where `e` is my caught exception) to save the error context before every `Logger.warning` or `Logger.error` or the like. Even worse, as a caller and especially someone inexperienced with `Logger`, there's no indication as to the lifetime of whatever is passed to `Logger.metadata`. Will it stay there until the next log, will it be included in every log from that point on? (spoiler alert, it's the latter, so you'd need to manually clear the crash reason in the metadata after the log).

tl;dr: this replaces

```elixir
try do
  raise "oops"
rescue
  e in RuntimeError -> 
      Logger.metadata(crash_reason: e)
      Logger.error("Something went wrong")
      Logger.metadata(crash_reason: nil)
end
```

```elixir
try do
  raise "oops"
rescue
  e in RuntimeError -> Logger.error("Something went wrong", crash_reason: e)
end
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -303,6 +303,36 @@ Any custom metadata fields will be added to the root of the message, so that you
 }
 ```
 
+You can also choose to log caught exceptions with a custom message.
```
Comment (AndrewDryga) [inline README.md:306]: I think this is a bit too verbose for repo's main `README.me` to include full documentation for all formatters, so let's not add it here.

Comment (AndrewDryga) [reply README.md:306]: Instead, we should add `:crash_reason` [here](https://github.com/Nebo15/logger_json/blob/master/lib/logger_json.ex#L72) as well-known metadata key supported across loggers, the documentation can look something like this:

```
* `:crash_reason` - accepts a tuple where the first element is the exception
    struct and the second is the stacktrace. For example: `Logger.error("Exception!", crash_reason: {e, __STACKTRACE__})`.
    Each formatter may encode it differently,
```

All the formatters should already support that.

File: lib/logger_json/formatters/elastic.ex [hunk 2]
```diff
@@ -95,6 +96,37 @@ defmodule LoggerJSON.Formatters.Elastic do
...
   ```
+
+  You can also choose to log caught exceptions with a custom message.
+  For example, after catching an exception with `try`/`rescue`:
+
+  ```elixir
+  try do
+    raise "oops"
+  rescue
+    e in RuntimeError -> Logger.error("Something went wrong", crash_reason: e)
```
Comment (AndrewDryga) [inline lib/logger_json/formatters/elastic.ex:107]: I think we can just document it to be a tuple that all providers support, so we won't need to add any special handling for this formatter:
```suggestion
    e in RuntimeError -> Logger.error("Something went wrong", crash_reason: {e, __STACKTRACE__)
```

File: README.md (hunk 1 above)
Comment (bvobart) [reply README.md]: Alright understandable, I've removed it from the ReadMe and added that `:crash_reason` documentation to `logger_json.ex`

File: lib/logger_json/formatters/elastic.ex (hunk 2 above)
Comment (bvobart) [reply lib/logger_json/formatters/elastic.ex]: I didn't know it was possible to use crash_reason like that! Thanks!

I was trying `Logger.error("Something went wrong", crash_reason: e)` with the implementation of the Elastic logger before this MR's changes and noticed that the test would just hang indefinitely.

I've removed my special code handling for it and cemented this crash_reason form in a test for the Elastic logger. I also left it in the documentation as it's quite useful info to know as someone implementing logging :)

**Review by AndrewDryga — APPROVED** [review]
Thank you ❤️
